### PR TITLE
[CI] CirlceCI: Use modularized docker containers

### PR DESCRIPTION
**Pull request by rleh**
from modm-ext/docker-modm-build#14

### CODE REVIEW

**Comment by salkinium** [comment]
I'd like to also unify these three commands into one, so that all three always get run, even if they fail. Otherwise you have to fix trailing whitespace, then wait 5mins, then fix docs, then wait 5mins, etc, which is dumb.

```yml
      - run:
          name: Check for Trailing Whitespace
          command: |
            python3 tools/scripts/rm_whitespace.py
      - run:
          name: Synchronize Documentation
          command: |
            python3 tools/scripts/synchronize_docs.py -d
      - run:
          name: Check for Unique Build Paths
          command: |
            python3 tools/scripts/examples_check.py examples
```

Into this? I think this still fails.

```yml
      - run:
          name: Sanity Checks: Whitespace, Docs, Build Paths
          command: |
            python3 tools/scripts/rm_whitespace.py
            python3 tools/scripts/synchronize_docs.py -d
            python3 tools/scripts/examples_check.py examples
```

I'll try this in another PR though, since this is off topic.

**Comment by rleh** [comment]
`when: always`?

**Comment by salkinium** [comment]
> `when: always`?

https://circleci.com/docs/2.0/configuration-reference/#the-when-attribute

Ah, that's obviously the right™ way. 🤦 

We should probably add this to all relevant steps, right?

**Comment by salkinium** [comment]
Multiple commands still abort obviously, so I pulled the relevant ones apart. Just a small optimization, but this works exactly like I want it too. EXCELLENT!

File: examples/avr/assert/main.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ modm_abandon(const modm::AssertionInfo &info)
 	serialStream << IFS(" failed!\nAbandoning...\n");
 #endif
 
-	Leds::setOutput();
+	Leds::setOutput();asdfasdfasdf
```
Comment (rleh) [inline examples/avr/assert/main.cpp]: Now remove all the `asdfasdfasdf` and we are ready to merge!

Comment (salkinium) [reply examples/avr/assert/main.cpp]: \*whip\*

**Review by rleh — COMMENTED** [review]
Approve :heavy_check_mark:

**Review by salkinium — APPROVED** [review]

**Pull request merged**